front/app/utils/cl-react-query/fetcher.ts

```diff
@@ -0,0 +1,142 @@
+import { API_PATH } from 'containers/App/constants';
+import { getJwt } from 'utils/auth/jwt';
+import { stringify } from 'qs';
+import { queryClient } from 'utils/cl-react-query/queryClient';
+import { isArray, isNil, omitBy } from 'lodash-es';
+import { reportError } from 'utils/loggingUtils';
+import { CLErrors } from 'typings';
+
+// FETCHER
+
+const jwt = getJwt();
+
+type Path = `/${string}`;
+interface Get {
+  path: Path;
+  action: 'get';
+  queryParams?: Record<string, any>;
+  body?: never;
+}
+interface Patch {
+  path: Path;
+  action: 'patch';
+  body: Record<string, any>;
+  queryParams?: never;
+}
+interface Post {
+  path: Path;
+  action: 'post';
+  body: Record<string, any>;
+  queryParams?: never;
+}
+interface Delete {
+  path: Path;
+  action: 'delete';
+  body?: Record<string, any>;
+  queryParams?: never;
+}
+
+type FetcherArgs = Get | Patch | Post | Delete;
+
+type BaseData = { id?: string; type: string };
+
+type BaseResponseData =
+  | { data: BaseData; included?: BaseData[] }
+  | { data: BaseData[]; included?: BaseData[] };
+
+function fetcher<TResponseData extends BaseResponseData>(
+  args: FetcherArgs
+): FetcherArgs['action'] extends 'delete'
+  ? null
+  : Promise<Omit<TResponseData, 'included'>>;
+
+async function fetcher({ path, action, body, queryParams }) {
+  const methodMap = {
+    get: 'GET',
+    patch: 'PATCH',
+    post: 'POST',
+    delete: 'DELETE',
+  };
+
+  // Remove unnecessary query parameters from object
+  const relevantQueryParams = omitBy(
+    queryParams,
+    (value) => isNil(value) || value === ''
+  );
```

Comment:
Could there be cases where an explicit empty parameter can be useful? There is a difference in an api between receiving `/some-method` or `/some-method?some-param=`, where `some-param` would be interpreted as the empty string.

If there is no specific reason to do this, I don't think we really want it (at least not for the empty sring case). If there is, could you add a comment to explain the why, rather than the what?

Reply:
There was indeed a case where if, for example, search was an empty string but was not explicitly excluded from the request, the query returned 0 results, so it really matters. I suspect this is also the reason query parameters were sanitized with the previous setup. We could also handle this at the component level, of course, but I think it's best to keep at least the component behaviour as backwards-compatible as possible as to not introduce subtle bugs. Will try to come up with a better comment for it, though. 👍

Reply:
Alright, agree that compatibility with current behavior is a good reason - for now